lib/turn.rb

```diff
@@ -0,0 +1,123 @@
+class Turn
+  attr_reader :player1, :player2, :spoils_of_war, :type
+
+  def initialize(player1, player2)
+    @player1 = player1
+    @player2 = player2
+    @spoils_of_war = []
+    #what is convention here?
+    if player1.deck.rank_of_card_at(0) == player2.deck.rank_of_card_at(0)\
+      && (player1.deck.cards[2] == nil || player2.deck.cards[2] == nil)
+      @type = :war
+    elsif player1.deck.rank_of_card_at(0) == player2.deck.rank_of_card_at(0)\
+      && player1.deck.rank_of_card_at(2) == player2.deck.rank_of_card_at(2)
+      @type = :mutually_assured_destruction
+    elsif player1.deck.rank_of_card_at(0) == player2.deck.rank_of_card_at(0)
+      @type = :war
+    else
+      @type = :basic
+    end
```

Comment:
Firstly, I would not put this complex conditional branching in the `initialize` method, I would set up a helper method for `@type` like so:
```
def initialize(player1, player2)
  ...
  @type = determine_type
end

def determine_type
  if @player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0) # ...etc
end
```

Also, if you check for the `mutually_assured_destruction` condition **first**, then `war`, I think this conditional logic could be simplified.